### PR TITLE
Display error message from URLError when the site's SSL is invalid

### PR DESCRIPTION
**Pull request by crazytonyli**
## Description

When connecting to a site which has invalid SSL certs, the error message is not super clear. This PR uses the error message came with URLError, which has more accurate information about the connection failure, for SSL handshake related errors.

Here is a comparison:

| Before  | After |
| ------- | ----- |
| Screenshot 2024-02-05 at 9 43 45 AM  | Screenshot 2024-02-05 at 9 52 35 AM  |

## Test Instructions

[This docker project](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/files/14158542/local-site.zip) can be use to set up a test site locally, which has invalid SSL certificate.
1. Download and unzip the file.
2. Go to the unzipped directory and execute `make run`. Requirements:
  - You'll need to have Docker Desktop or equivalent running.
  - The 80 port on your Mac needs to be available for the docker container.
4. Open Safari from a device (which needs to be in the same network as your Mac), or an iOS simulator.
5. Enter `https://<your-mac-hostname>.local` to test the site is accessible and it has an invalid SSL certificate (Safari should warn you about it).
6. Open Jetpack/WordPress iOS app from the same deivce or simulator.
7. Use "Enter your existing site address" option and enter the same https address as above.
8. Checkout the error message.

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteAddressViewController.swift [hunk 1]
```diff
@@ -481,7 +481,15 @@ private extension SiteAddressViewController {
                         return
                     }
 
-                    return self.displayError(message: Localization.nonExistentSiteError, moveVoiceOverFocus: true)
+                    var message: String?
+
+                    // Use `URLError`'s error message (which usually contains more accuruate description), if the
```
Comment (mokagio) [inline WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteAddressViewController.swift]:
```suggestion
                    // Use `URLError`'s error message (which usually contains more accurate description), if the
```

**Review by mokagio — APPROVED** [review]
Have you considered using the `localizedDescription` if the error is a `URLError` regardless of whether it was an SSL issue?

_Thank you for the before/after images_ 😄

**Comment by crazytonyli** [comment]
> Have you considered using the localizedDescription if the error is a URLError regardless of whether it was an SSL issue?

We probably don't want to display all `URLError`s to user. For example, `URLErrr.timeout` even though it's a reasonable message for user to see. But in this particular case here, "site is not reachable" is probably more appropriate than "the request has timed out". Plus, there are [so many different URLError cases](https://developer.apple.com/documentation/foundation/urlerror/code), I'm being a bit cautious here, to only select SSL ones because it's truly an error that occurs during HTTP requests.

**Comment by mokagio** [comment]
Thank you for expanding on the error management rationale @crazytonyli

**Pull request merged**